improver/psychrometric_calculations/hail_size.py

```diff
@@ -485,53 +498,49 @@ def get_hail_size(
         horizontal_rounded = np.around(horizontal / 5, decimals=0) - 1
         vertical_rounded = np.around(vertical * 2, decimals=0)
 
-        # flattens array's so they can later be accessed
-        horizontal_flat = horizontal_rounded.flatten(order="C")
-        vertical_flat = vertical_rounded.flatten(order="C")
-        wet_bulb_zero_flat = wet_bulb_zero.flatten(order="C")
-
-        hail_size_list = []
         # clips index values to not be longer than the table
-        vertical_clipped = np.clip(vertical_flat, None, len(lookup_table) - 1)
-        horizontal_clipped = np.clip(horizontal_flat, None, len(lookup_table[0]) - 1)
-
-        for vert, hor, wbz in zip(
-            vertical_clipped, horizontal_clipped, wet_bulb_zero_flat
-        ):
-            if min(hor, vert) < 0 or not (vert and hor) or wbz > 4400:
-                hail_size = 0
-            else:
-                hail_size = lookup_table[int(vert)][int(hor)]
-            if wbz >= 3300:
-                hail_size = self.updated_hail_size(hail_size, wbz)
-            hail_size_list.append(hail_size)
-
-        hail_size = np.reshape(hail_size_list, shape, order="C")
+        vertical_clipped = np.clip(vertical_rounded, None, len(lookup_table) - 1)
+        horizontal_clipped = np.clip(horizontal_rounded, None, len(lookup_table[0]) - 1)
+
+        vertical_clipped = np.ma.where(
+            (vertical_rounded >= 0) & (horizontal_rounded >= 0), vertical_clipped, 0
+        ).filled(0)
+        horizontal_clipped = np.ma.where(
+            (vertical_rounded >= 0) & (horizontal_rounded >= 0), horizontal_clipped, 0
+        ).filled(0)
```

Comment:
I haven't done anything with this. We need a test that has NaN temperature and RH values in the bottom one or two pressure levels to simulate the sort of data we get from StaGE (these levels can fall below the model orography, so are undefined). It may be that this "filled" method is needed to keep these cases happy.

Reply:
I think I've fixed this now